src/data_structures/fenwick.md

```diff
@@ -60,20 +65,21 @@ def increase(int i, int delta):
 
 The function `sum` works as follows:
 
-1. first, it adds the sum of the range $[g(r), r]$ (i.e. $T[r]$) to the `result`
-2. then, it "jumps" to the range $[g(g(r)-1), g(r)-1]$, and adds this range's sum to the `result`
-3. and so on, until it "jumps" from $[0, g(g( \dots g(r)-1 \dots -1)-1)]$ to $[g(-1), -1]$; that is where the `sum` function stops jumping.
+1. First, it adds the sum of the range $[g(r), r]$ (i.e. $T[r]$) to the `result`.
+2. Then, it "jumps" to the range $[g(g(r)-1), g(r)-1]$ and adds this range's sum to the `result`.
+3. This continues until it "jumps" from $[0, g(g( \dots g(r)-1 \dots -1)-1)]$ to $[g(-1), -1]$; this is where the `sum` function stops jumping.
 
-The function `increase` works with the same analogy, but "jumps" in the direction of increasing indices:
+The function `increase` works with the same analogy, but it "jumps" in the direction of increasing indices:
 
-1. sums of the ranges $[g(j), j]$ that satisfy the condition $g(j) \le i \le j$ are increased by `delta` , that is `t[j] += delta`. Therefore we updated all elements in $T$ that correspond to ranges in which $A_i$ lies.
+1. Sums of the range $[g(j), j]$ which satisfy the condition $g(j) \le i \le j$ are increased by `delta`; that is, `t[j] += delta`.
```

Comment:
I think it still should be `ranges` instead of `range`? This is talking about a group of such ranges, and the sum is increased for each such range.

Reply:
I was trying to word it as something along the lines of sums of each range (for arbitrary `j` under the mentioned constraints).  I agree that the wording is a bit clunky, but I'm not sure that the use of `ranges` makes it less ambiguous.  Maybe something like `The sum for each range of the form ... is increased by ...` would make more sense?

Reply:
Yes, I think it'd be better, please proceed. Thanks!

Reply:
Alright, I've made a commit which should (hopefully) properly reflect this.